keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt

```diff
@@ -66,17 +66,13 @@ class ArtifactListener(
   @EventListener(ArtifactRegisteredEvent::class)
   fun onArtifactRegisteredEvent(event: ArtifactRegisteredEvent) {
     val artifact = event.artifact
-    if (artifactRepository.isRegistered(artifact.name, artifact.type)) {
-      log.debug("Artifact {} is already registered", artifact)
-    } else {
-      log.debug("Registering artifact {}", artifact)
-      artifactRepository.register(artifact)
-    }
+    artifactRepository.register(artifact)
```

Comment:
Is this call just a no-op if the artifact is registered already?

Reply:
It's an update call if the artifact is registered so that we can handle changes in docker sorting. If this gets sql load intensive we can optimize this.

Reply:
Got it.